apps/src/blockly/addons/cdoSerializationUtils.js

```diff
@@ -0,0 +1,109 @@
+import BlockSvgUnused from './blockSvgUnused';
+
+const SORT_BY_POSITION = true;
+
+/**
+ * Converts XML serialization to JSON using a temporary unrendered workspace.
+ * @param {xml} xml - workspace serialization, current/legacy format
+ * @return {json} stateToLoad - modern workspace serialization
+ */
+export function convertXmlToJson(xml) {
+  const tempWorkspace = new Blockly.Workspace();
+
+  // domToBlockSpace returns an array of "block" objects with the following property:
+  //   blockly_block: the actual block object created by Blockly
+  //   x: the x attribute found in <block/> element
+  //   y: the y attribute found in <block/> element
+  const xmlBlocks = Blockly.Xml.domToBlockSpace(tempWorkspace, xml);
+
+  const stateToLoad = Blockly.serialization.workspaces.save(tempWorkspace);
+
+  // Create a map of ids (key) and blocks (value).
+  const blockIdMap = new Map();
+  stateToLoad.blocks.blocks.forEach(block => {
+    blockIdMap.set(block.id, block);
+  });
+
+  addPositionsToState(xmlBlocks, blockIdMap);
+  tempWorkspace.dispose();
+  return stateToLoad;
+}
+
+/**
+ * Converts XML serialization to JSON using a temporary unrendered workspace.
+ * @param {Array<Object>} xmlBlocks - an array of "block" objects containing a block and x/y coordinates
+ * @param {Map<String, Object>} blockIdMap - a map of ids (keys) and blocks (values)
+ */
+function addPositionsToState(xmlBlocks, blockIdMap) {
+  xmlBlocks.forEach(xmlBlock => {
+    const block = blockIdMap.get(xmlBlock.blockly_block.id);
+    if (block && xmlBlock.x && xmlBlock.y) {
```

Comment:
Not that I can find! But I'm happy to adjust so that each is set separately.

Reply:
In the case above for `when hit the ground`, would it be possible to add a cushion for the y value? Or maybe this is a way to remind levelbuilders to add both x and y coordinates? 
Can you remind me for which Google Blockly labs is there a reorganization of the blocks and levelbuilders are unable to set the blocks to a certain location? I remember Emma asking about this.

Reply:
Emma was asking about Sprite Lab, where it looks like we don't currently save block X/Y coordinates. This will start to happen by default once it's on Google Blockly, fortunately.

I'll look into the cushion idea. It will complicate the logic a bit. If both coordinates are 0, we let the cursor position it. So we'd only want to add the cushion to the unspecified value if the other was specified.